src/crypto/PublicKey.ts

```diff
@@ -248,6 +253,63 @@ export class AccountLinkedPublicKeyV1
     return this.keyBytes
   }
 
+  // Return the extracted address from the SIWE signature checking role and keys, but not domain
+  private getSIWELinkedAddressWithoutDomainCheck(
+    role: AccountLinkedRole
+  ): string {
+    // The criteria for successful SIWE-signed keys with roles are:
+    // 1. The text must be a valid SIWE for the expected domain (TBD)
+    // 1b. Check expiry (TBD)
+    // 2. The resources must contain the "https://[role]/(keybytes as hex, uncompressed){64}" with no
+    //    extraneous characters. We include the curve to prevent reuse across curves.
+    // 3. Check the signature
+
+    // Only works if we have a SIWE signature, we might have multiple types which is okay
+    if (!this.siweSignature) {
+      throw new Error('No SIWE signature')
+    }
+    // Assume v1 for now, would need to switch/case on version in the future
+    const siweSignature = this.siweSignature
+
+    // Rule 1: Parse the text as SIWE
+    // convert text bytes to a string as utf-8 encoded
+    const textUtf8 = new TextDecoder().decode(siweSignature.text)
+    const siwe = new SiweMessage(textUtf8)
+    // TODO: add this domain check when it makes sense
+    // if (siwe.domain !== domain) {
+    //   console.log(`Expected domain ${domain} but got ${siwe.domain}`)
+    //   return false
+    // }
+    // Rule 2: Check the resources
+    const resources = siwe.resources || []
+    const roleString =
+      SIWEWalletAccountLinkSigner.accountLinkedSIWERoleRequestText(role)
+
+    const expectedResource = `https://xmtp.org/siwe/${roleString}/secp256k1/${bytesToHex(
```

Comment:
This string is coded in two locations, here and the validation step. Unifying these will reduce the number of updates needed, and be easier to maintain

Reply:
Yup, will consolidate